src/Access/SettingsProfileElement.h

```diff
@@ -19,6 +19,10 @@ class AccessControl;
 struct SettingsProfileElement
 {
     std::optional<UUID> parent_profile;
+    /// parent_profile_xml_name is set only when users are configured via XML
+    /// and is used for user-friendly error messages.
+    std::optional<String> parent_profile_xml_name;
```

Comment:
`SettingsProfileElement` is used not only with `users.xml`, so it seems it would be better to check that all parent profiles exist in `parseSettingsProfiles()` in UsersConfigAccessStorage, without introducing this new field.

Reply:
For example we can make a set of all profile names in `parseSettingsProfiles()` first and then pass this set to `parseSettingsProfile()`.